app/src/main/java/de/tum/in/tumcampusapp/component/tumui/calendar/widget/TimetableWidget.kt

```diff
@@ -0,0 +1,152 @@
+package de.tum.`in`.tumcampusapp.component.tumui.calendar.widget
+
+import android.app.AlarmManager
+import android.app.PendingIntent
+import android.appwidget.AppWidgetManager
+import android.appwidget.AppWidgetProvider
+import android.content.ComponentName
+import android.content.Context
+import android.content.Intent
+import android.net.Uri
+import android.widget.RemoteViews
+import de.tum.`in`.tumcampusapp.R
+import de.tum.`in`.tumcampusapp.component.tumui.calendar.CalendarActivity
+import org.joda.time.DateTime
+import org.joda.time.format.DateTimeFormat
+import java.util.*
+
+class TimetableWidget : AppWidgetProvider() {
+
+    override fun onUpdate(context: Context, appWidgetManager: AppWidgetManager, appWidgetIds: IntArray) {
+        updateAppWidgets(context, appWidgetManager, appWidgetIds)
+        setAlarm(context)
+    }
+
+    override fun onDeleted(context: Context, appWidgetIds: IntArray) {
+        // When the user deletes the widget, delete the associated setting from the database.
```

Comment:
The comment doesn't fit the method content. Should we delete something from the database here? If not, this method can be removed.